hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/connector/ReadWithPartitionIteratorP.java

```diff
@@ -317,12 +341,85 @@ public void init(@Nonnull Context context) {
         @Override
         public void init(@Nonnull Context context) {
             partitionToIterator = iteratorSupplier.apply(context.jetInstance().getHazelcastInstance());
+            migrationWatcher = new MigrationWatcherImpl(context.jetInstance().getHazelcastInstance());
         }
 
         @Override @Nonnull
         public List<Processor> get(int count) {
-            return getProcessors(count, ownedPartitions, partitionToIterator);
+            return getProcessors(count, ownedPartitions, partitionToIterator, migrationWatcher);
+        }
+
+        @Override
+        public void close(@Nullable Throwable error) {
+            if (migrationWatcher != null) {
+                migrationWatcher.deregister();
+            }
         }
+    }
+
+    interface MigrationWatcher {
```

Comment:
You already removed it :)

Reply:
weird, this was my pending review and rather than publishing I fixed it, but now it seems to have published it without my approval because I made another comment somewhere :)